File: DP/9-palindrome-number/palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):

        """
        :type x: int
        :rtype: bool
        """
        if(x < 0):
          return False
        i=x 
        ref = 0
        while(i!=0):  
            d = i%10;
            ref = ref *10+d;
            i=i//10;
            print(i)

        return x==ref
        

File: DP/9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0)
          return false;
   int ref = 0;
         for(int i =x;i!=0 ; i/=10)
         {
             int d = i%10;
              ref = ref *10+d;
         } 
         if(x==ref)
           return true;
           else return false;
    }
}